CONTRIBUTING.md

```diff
@@ -0,0 +1,106 @@
+
+  Over the years we've seen many PRs targeting areas, which we didn't plan to expand further at the time.  In many of these these cases we had to say `no` to those PRs and close them. That, obviously, is not a great outcome for us. And it's especially bad for the contributor, as they've spent a lot of effort preparing the change.
+  To resolve this problem, we've decided to separate a bucket of issues, which would be great candidates for community members to contribute to. We mark these issues with the `help wanted` label. [help wanted](https://github.com/dotnet/fsharp/labels/help%20wanted)
+
+  Within that set, we have additionally marked issues that are good candidates for first-time contributors. Here: [Good first issue](https://github.com/dotnet/fsharp/labels/good%20first%20issue)
+
+  If you would like to make a contribution to an area not documented here, first open an issue with a description of the change you would like to make and the problem it solves so it can be discussed before a pull request is submitted.
+
+### The primary customers of the F# repository are users of the dotnet SDK, Visual Studio, Rider and Ionide. At all times their experience is paramount in our mind.
+
+  We are very accepting of community pull requests, there are however, a set of firm considerations that we hold to when reviewing PRs and determining what to merge.  These have been developed over the years to maintain the quality of the product and the experience that F# developers have when installing and upgrading the dotnet SDK and Visual Studio.
+
+- Does the change fix something that needs fixing, is there an issue, does the issue indicate a real problem?
+- Does the change improve the readability of something that needs improvement?
+- Does the change add a feature that is approved for adding?
+- Does the code match or improve of the existing codebase?
+- Is the performance improvement measured and can regressions be identified?
+- Will our existing customers be able to without effort upgrading the **Major** release of an SDK or VS?
+- Will our existing customers be able to without effort upgrading the **Minor** release of an SDK or VS?
+- Does it break VS, Rider or Ionide?
+- Does it have adequate testing?
+- Do all existing tests run unmodified?
+
+In general answers to the above should be **Yes**.  A **No** to any of them is not disqualifying of the PR, however a no answer will need an explanation and a discussion.
+ 
+There are additional considerations
+- Is the risk of accepting this change High or even Medium
+  For an acceptable PR with a high risk, we will definitely need to discuss mitigations for the risk.  A decision to upgrade the SDK or VS needs to be always low risk for our customers, they have businesses to run, they don't want to have to deal with our - risky behavior.  We may defer or delay risky PRs into a later release or abandon it.
+- Is the change as small as possible
+- Should it be chopped up into smaller chunks
```

Comment:
```suggestion
- Should it be chopped up into smaller, yet independently valuable and releasable to production, chunks
```

Reply:
I don't think "releasable" is a proper word tbh :D. I would word it simply "Should it be chopped up into smaller, independent chunks"